tests/Auth0.ManagementApi.IntegrationTests/UsersTests.cs

```diff
@@ -306,6 +306,44 @@ public async Task Can_update_user_metadata()
 
         }
 
+
+        [Fact]
+        public void Can_read_user_metadata_as_type()
+        {
+            // create a user instance
+            var user = new User
+            {
+                Email = $"{Guid.NewGuid():N}{TestingConstants.UserEmailDomain}",
+                EmailVerified = true
+            };
+            var customAppMetadata = new CustomMetadata
+            {
+                Amount = 42,
+                Name = "A name"
+            };
+            var customUserMetadat = new CustomMetadata
+            {
+                Amount = 43,
+                Name = "One off"
+            };
+            
+            // set metadata
+            user.SetAppMetadata(customAppMetadata);
+            user.SetUserMetadata(customUserMetadat);
+
+            // read metadata
+            var appMetadata = user.GetAppMetadata<CustomMetadata>();
+            var userMetadata = user.GetUserMetadata<CustomMetadata>();
+            
+            Assert.NotNull(appMetadata);
+            Assert.Equal(appMetadata.Amount, customAppMetadata.Amount);
+            Assert.Equal(appMetadata.Name, customAppMetadata.Name);
+
+            Assert.NotNull(userMetadata);
+            Assert.Equal(userMetadata.Amount, customUserMetadat.Amount);
+            Assert.Equal(userMetadata.Name, customUserMetadat.Name);
+        }
```

Comment:
Can we update this test to use Set, then call Auth0 to create the user, then use Get to verify it gets serialized and send correctly?